doc/grammars/type.abnf

```diff
@@ -85,18 +85,18 @@ ConstantExpr
 	/ ConstantFetch *ByteHorizontalWs
 
 ConstantFloat
-	= ["-"] 1*ByteDecDigit "." *ByteDecDigit [ConstantFloatExp]
-	/ ["-"] 1*ByteDecDigit ConstantFloatExp
-	/ ["-"] "." 1*ByteDecDigit [ConstantFloatExp]
+	= ["-"] 1*ByteDecDigit *("_" 1*ByteDecDigit) "." [1*ByteDecDigit *("_" 1*ByteDecDigit)] [ConstantFloatExp]
+	/ ["-"] 1*ByteDecDigit *("_" 1*ByteDecDigit) ConstantFloatExp
+	/ ["-"] "." 1*ByteDecDigit *("_" 1*ByteDecDigit) [ConstantFloatExp]
 
 ConstantFloatExp
-	= "e" ["-"] 1*ByteDecDigit
+	= "e" ["-"] 1*ByteDecDigit *("_" 1*ByteDecDigit)
```

Comment:
please see https://github.com/phpstan/phpstan-src/pull/2358#discussion_r1178091844, `+` before exponent should be supported, for consistency, also before any number.

Reply:
yep, it makes sense to support the same range of expressions as PHP does. But it feels out of scope of this PR; feel free to follow up with a fix :)

Reply:
I will, this PR is merged, thus there will be no conflicts... One question, why is the grammar defined on two places - doc/grammars/type.abnf and src/Lexer/Lexer.php, is one file generated from the other one?

Reply:
`doc/grammars/type.abnf` is mostly for documentation purposes and it's very likely out of sync. There's also FuzzyTest using it but I'm not sure about its purpose.